### ISSUE TITLE
[Feature Request]: Arbitrary select assets for mass update

### ISSUE THREAD

**Issue by MWR-Napavine** [grokability/snipe-it#11241]
### Is your feature request related to a problem? Please describe.

Scenario: I need to update 15 assets which do not naturally relate to each other. Perhaps they are all different assets that have been moved into the same room. I want to update them all without going into each asset individually to edit. Lets say I want to change the Default Location and Notes fields. Maybe even a custom field. 

Currently, the system will allow you to search for an asset, select it, then search for additional assets to select, before using the Edit button to modify the multiple assets. But this requires multiple clicks: Scan with hand scanner, then select the asset with the mouse, then return the curser back to the scanning field.

### Describe the solution you'd like

This could be fixed in two ways: A modification to the current interface, or a whole new interface.

Modification method: Have the current implementation perform the search **and select any exact match** when the search field is used with an enter. Then clear the search field (or alternatively, keep the search field full, but text fully selected so it is easily overwritten) so the next scan can take place. In addition, show an indication that "X Assets selected. _Show List_" Have that Show list open a list of selected Assets Tags, each with an X so it can be easily de-selected. Then the Edit button would go to the existing Multiple-Asset editing page.

New Interface method: Similar the Bulk Check-in pages, but instead of checking-in the assets, it simply builds that list of "selected" assets than can then be mass edited.

### Describe alternatives you've considered

An alternative might be a simplified way to pull in an arbitrary list. How about a simple open mult-line field that allows you to enter asset tags, 1-per-line, and then you can proceed to the mass edit screen?

### Additional context

One last thing. Every "Mass Edit" screen should allow me to see which asset(s) I am editing. That way I can verify that the correct assets are getting the edit(s).

**Comment by snipe** [comment]
What version of snipe-it are you using?

**Comment by MWR-Napavine** [comment]
Hosted by you, version 6.0.1.


--Matthew Ross
Napavine School District <http://www.napavineschools.org/>
***@***.***
Technology Department
Desk: 360-334-7907

**Comment by snipe** [comment]
So I think I'm a little unclear here though - the current UI allows bulk editing. Not custom fields, since multiple assets that belong to different models often have different custom fields, and that UI could be very confusing - but you can change locations, etc. You can also do this by category, model, etc. 

We don't currently show the list of assets because if you've selected 100+ assets, that page gets extraordinarily long to the point of not really being usable.

**Comment by MWR-Napavine** [comment]
The difficulty in bulk editing isn't that the system doesn't allow it, it's more that it's difficult to quickly do it with arbitrary, not-easily searchable assets.

So, my example is a Classroom: Items were moved over the Summer break between classrooms, and I now need to update multiple fields for the 15 assets I have in front of me. They need to have their default location changed, and I want to add something to the Notes field.

Currently, I have to Search by asset tag, select, search again, select, search again... I would like to be able to do all the selecting quickly with a standard barcode scanner. Zap, zap, zap... I've built my list. Then I can mass edit.

As for the "100+ assets is too many to be usable", I agree... which is why I suggest that the list of assets be in a collapsed list. (I was trying to describe a collapsed list without using the vernacular, because I couldn't remember the word for it.) Then have a "show the assets" option to let me see the list if I want/need to. Maybe a checkbox that says "Always show the asset list" could be there too, for those of us who always want it. (I know, we're insane!)

Thanks for everything.

**Comment by snipe** [comment]
How are your classrooms set up? Are they locations? If so, you can just select the assets via the location listing and bulk edit them there, no?

**Comment by MWR-Napavine** [comment]
Ah, you're assuming the source location is the same... In my scenario, the items came from multiple source locations and are now being moved to a new target location. This kind of thing does happen.

Other examples that might benefit from the same arbitrary selection process: Updating a list of damaged items in a fire. Updating the warranty info for a list of devices. Putting a list of Assets up for a surplus auction.

Maybe a different way to do this is a "build a list" feature... Even if it's temporary, it would be nice sometimes to have a list of assets for different reasons. Example lists could be: "Tables used for ceremony" or "Johny's tools on loan" or "Computers marked for upgrades." Then using this built list, I can mass-edit... and disband the list if I want to. 

Thanks for considering my feature requests.

**Comment by MWR-Napavine** [comment]
Thinking of this more... I do like the "build a list" feature more... IT also allows me to make a report of specific items, even if that list of items is arbitrary. This would be useful for a random audit, for example.

**Comment by snipe** [comment]
Would the idea of "tags" be something that you could work with?

**Comment by MWR-Napavine** [comment]
Yes, Tags would function like my custom lists idea, as long as I can bring up that list of tagged items, mass edit from there, and export reports.

But if it is difficult to tag multiple items, it doesn't solve the original issue.  If I have to go into each item and add the tag, or use the search feature to select each item individually to mass-add the tag, it's the same problem.

Here's another  real-life example I'm currently experiencing: I've created a new Status called "Retired". It's for my items that are no longer in use, but are not yet surplussed. They live in my storage, and could be pulled out of retirement and back into use if necessary. I'm working to change a large number of Deployable items so that they are now Retired... But the items are not sequential, not the same models, and some are missing... so I have to make a list. Right now, that list is going into a spreadsheet which I will then later have to edit in Snipe-IT. There is no way to quickly change these items to a different Status.

I currently am aware of 3 options to proceed: 1) Individually open each item, change the status, and hit save. 2) Search for each item using the "All Assets" list, Select it, Repeat search and select for each item, then do a mass edit, or 3) Download the relative data as a CSV, do some spreadsheet matching vlookup magic so I get the right items, change the field correctly, then re-upload the CSV as an import that hopefully doesn't mess anything up. None of these options are as simple as I wish them to be.

I think the simplest solution is to allow me to import a list of items to select. It would just be an open text field that would parse for Asset IDs, 1 per line. The resulting search would then take me to the mass edit page for those items.

**Comment by SoreThumb1** [comment]
I completely understand what @MWR-Napavine is asking for and would also benefit greatly from this type of feature. As of now, when trying to edit multiple assets at once, it isn't a clean and straightforward task. The checkboxes method is only useful when selecting a handful of items that are all on the screen at once, or editing an entire inventory where you can just select all. But when you are picking and choosing a bulk of assets to edit, I find myself going back and forth a lot, doing them little at a time.

Scenario: I'm trying to move 50 of 100 assets to another location. I want to edit the assets that I'm moving to reflect the change in location and status. What's the best way to do that?

Some sort of list importer that matches list items to assets by tag or serial#, then you can manually add additional assets if they weren't automatically detected, then you can edit them.

**Comment by snipe** [comment]
@SoreThumb1 The asset importer has an update mode already.

> Scenario: I'm trying to move 50 of 100 assets to another location. I want to edit the assets that I'm moving to reflect the change in location and status. What's the best way to do that?

I would go to the location you're moving them *from*, go to the asset listing, bulk edit them.

<img width="1270" alt="Screen Shot 2022-07-02 at 4 21 35 PM" src="https://user-images.githubusercontent.com/197404/177018789-d4f7c8e8-cd88-44d1-b5d9-1b96fd16e0b1.png">

**Comment by SoreThumb1** [comment]
@snipe 
Thank you for the suggestion, however, that would still result in me manually ticking the checkboxes of the ones I want. I wouldn't be able to 'select all' if I were only editing a portion of those assets.

**Comment by MWR-Napavine** [comment]
I'm back revisiting this old request. Looks like there might be a UI solution to this already in Snipe-IT:

In Assets -> Bulk Checkout, the Assets field on that page allows you to insert a large number of items check out. You could use this exact same "Asset Picker" UI, but then send the user to the Bulk Edit page:
![SnipeCanPickArbitrary](https://user-images.githubusercontent.com/48335860/196297907-33778add-c137-4586-90b4-d6af76b53c0a.gif)

This asset picker UI would need to allow pasting in lists of items. Currently it does not:
![SnipeCantPaste](https://user-images.githubusercontent.com/48335860/196297942-04de6e36-bf5c-469b-b350-4baf4d3936eb.gif)

Of course, if a different solution is already being implemented, I'm exited to see the update.

**Comment by MWR-Napavine** [comment]
Checking in on this. There is still no way to do checkout/editing of arbitrary items (unless I'm missing something). Current version is 7.0.10.

**Comment by snipe** [comment]
You'd narrow down that list via model. category, tag search, name search, or any of the the other ways we allow you to narrow down results.
